File: Малина/Автопилот/Автопилот_квадрокоптер/math_model.py
class TMath_model:
    import numpy as np
    """docstring for ."""
    def __init__(self, X_come):
        self.X = X_come
        self.X_for_analiz=self.X
    def write_result(self):
        print(X)



    def rightpart(self,t):

        dX=self.f(t,self.X,self.U(t))
        return dX
    def DU(self,t):
        self.t0=self.t1
        self.t1=t
        dt=self.t1-self.t0
        self.X=self.X+self.rightpart(t)*dt
    def analiz_vector(X_came_for_analiz,t_come):
        self.X_for_analiz=X_came_for_analiz
        self.time_bort=t_come
        DU(self.time_bort)
        dx_analiz=X_came_for_analiz-self.X
        return dx_analiz





    def f(self,t):
        F_out=
        return F_out
    def U(self,t):
        U_out=[]
        return U_out

File: TestQT_windows/mathmodel.cpp
#include "mathmodel.h"

mathmodel::mathmodel(std::vector<float> X_come) {

  X = X_come;
  X_for_analiz = X;
}

void mathmodel::write_result() {}

//  правые части
std::vector<float> mathmodel::rightpart(float t) {

  std::vector<float> dX = {0, 0, 0, 0, 0, 0, 0, 0};
  return dX;
}

// интегрирование
std::vector<float> mathmodel::DU(float t) {
  t0 = t1;
  t1 = t;
  float dt = t1 - t0;
  std::vector<float> dx;
  dx = rightpart(t);

  for (int i = 0; i < 8; i++) {

    X[i] = X[i] + dx[i] * dt;
  }

  return X;
}
//анализ вектора

std::vector<float>
mathmodel::analiz_vector(std::vector<float> X_came_for_analiz, float t_come) {
  X_for_analiz = X_came_for_analiz;
  float time_bort = t_come;
  DU(time_bort);
  std::vector<float> dx_analiz(X_came_for_analiz.size());
  for (int i = 0; i < X_came_for_analiz.size(); i++) {
    dx_analiz[i] = X_came_for_analiz[i] - X[i];
  }
  return dx_analiz;
}

/*float* mathmodel::f(float t):
    F_out=
    return F_out
*/
std::vector<float> mathmodel::U(float t) {
  std::vector<float> U_out = {0, 0, 0, 0, 0, 0, 0, 0};
  return U_out;
}
